test_crates/inherent_method_now_doc_hidden/old/src/doc_hidden_struct_inherent_method_now_doc_hidden.rs

```diff
@@ -0,0 +1,15 @@
+// Tests false positives. If the struct is #[doc(hidden)], changing hidden state of methods should
+// have no effect.
```

Comment:
Another interesting test case variant, if you choose to add it, would be a further tweak of this one.

Instead of having the struct start off as `#[doc(hidden)]` originally in the `old` version, only make it `#[doc(hidden)]` in the `new` version alongside hiding its methods as well. That should _also_ not trigger this lint, though it will trigger the `struct_now_doc_hidden` lint instead.

Reply:
Just pushed, hopefully passes CI

Reply:
Looks like it passed! I'm happy with this, and I can merge it as soon as you're happy with it too.

Thanks for putting it together!

Reply:
I'm ready to merge!